src/ffmpeg_subscriber.cpp

```diff
@@ -51,66 +83,131 @@ void FFMPEGSubscriber::subscribeImpl(
 #endif
 }
 #endif
-
 void FFMPEGSubscriber::initialize(rclcpp::Node * node, const std::string & base_topic)
 {
   node_ = node;
-  // Declare Parameters
-  uint ns_len = node->get_effective_namespace().length();
+  uint ns_len = node_->get_effective_namespace().length();
   std::string param_base_name = base_topic.substr(ns_len);
   std::replace(param_base_name.begin(), param_base_name.end(), '/', '.');
-  param_namespace_ = param_base_name + "." + getTransportName() + ".map.";
-  // create parameters from default map
-  for (const auto & kv : ffmpeg_encoder_decoder::Decoder::getDefaultEncoderToDecoderMap()) {
-    const std::string key = param_namespace_ + kv.first;
-    rclcpp::ParameterValue v;
+  param_namespace_ = param_base_name + "." + getTransportName() + ".";
+
+  for (const auto & p : params) {
+    declareParameter(node, param_base_name, p);
+  }
+}
+
+void FFMPEGSubscriber::declareParameter(
+  rclcpp::Node * node, const std::string & base_name, const ParameterDefinition & definition)
+{
+  // transport scoped parameter (e.g. image_raw.compressed.format)
+  const auto v = definition.declare(node, base_name, getTransportName());
+  const auto & n = definition.descriptor.name;
+  if (n == "av_options") {
```

Comment:
@berndpfrommer meta: this is a bit peculiar, we have code to allow for an arbitrary set of parameters but then we hardcode their names and limit the set here? :thinking: